File: app/src/main/java/com/ayatkhraisat/mvp_example/di/scopes/ViewScope.java
package com.ayatkhraisat.mvp_example.di.scopes;

import javax.inject.Scope;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

/**
 * Created by Ayat Khriasat on 14,December,2018 at 7:33 PM
 * Email: ayatzkhraisat@gmail.com
 * Project: MvpExample
 */

@Scope
@Retention(RetentionPolicy.RUNTIME)
public @interface ViewScope {
}

File: app/src/main/java/com/ayatkhraisat/mvp_example/di/scopes/ViewScope.kt
package com.ayatkhraisat.mvp_example.di.scopes

import java.lang.annotation.RetentionPolicy
import javax.inject.Scope
import kotlin.annotation.Retention

/**
 * Created by Ayat khraisat  on 2/12/19
 * akhraisat@blessedtreeit.com
 * <p>
 * Project Name: MVP_Architecture
 * <p>
 * Blessed Tree IT
 */
@Scope
@Retention(AnnotationRetention.RUNTIME)
annotation class ViewScope
